### PR TITLE
DT-1179: ✨ Add in retrieval of nomis users by email address

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -6,7 +6,11 @@ orbs:
 
 jobs:
   validate:
-    executor: hmpps/java
+    executor:
+      name: hmpps/java
+      environment:
+        _JAVA_OPTIONS: -Xmx512m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -XX:ParallelGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false
```
Comment (petergphillips) [inline .circleci/config.yml:12]: was getting out of memory on compiling test kotlin

File: src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/NomisUserService.kt [hunk 2]
```diff
@@ -1,17 +1,35 @@
...
+abstract class NomisUserService(private val staffUserAccountRepository: StaffUserAccountRepository,
+                                private val userRepository: UserRepository) {
   fun getNomisUserByUsername(username: String): Optional<NomisUserPersonDetails> =
-      staffUserAccountRepository.findById(StringUtils.upperCase(username))
+      staffUserAccountRepository.findById(username.toUpperCase())
+
+  fun getNomisUsersByEmail(email: String): List<NomisUserPersonDetails> {
+    val emailLowered = email.toLowerCase()
+
+    val allNomisInAuthUsernames = userRepository.findByEmailAndSourceOrderByUsername(emailLowered, nomis)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/oauth2server/security/NomisUserService.kt:21]: could be that the user has a verified email address in auth

File: .circleci/config.yml [hunk 3]
```diff
@@ -7,6 +7,9 @@ orbs:
 jobs:
   validate:
     executor: hmpps/java
+    environment:
+      _JAVA_OPTIONS: -Xmx512m -XX:ParallelGCThreads=2 -XX:ConcGCThreads=2 -XX:ParallelGCThreads=2 -Djava.util.concurrent.ForkJoinPool.common.parallelism=2 -Dorg.gradle.daemon=false
```
Comment (petergphillips) [inline .circleci/config.yml:11]: default is 256m

**Review by Mjwillis — APPROVED** [review]
LGTM

**Review by andymarke — DISMISSED** [review]
LGTM

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**